src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -1955,6 +1978,11 @@ private synchronized static void propagateLocalIndexStatus(String keyspace, Stri
             else
                 states.put(keyspaceIndex, status);
 
+            // Make sure to check this after peerIndexStatus is populatedm but before the JSON is built: we don't need
+            // the latter if there's no Gossiper (as in CNDB), so we can avoid the related CPU and memory usage.
+            if (!Gossiper.instance.isEnabled())
+                return;
```

Comment:
I've tested this last change in the context of CNDB-12425 and it works like a charm :)